udata/core/dataset/models.py

```diff
@@ -416,6 +459,48 @@ def update_resource(self, resource):
     def community_resources(self):
         return self.id and CommunityResource.objects.filter(dataset=self) or []
 
+    @cached_property
+    def json_ld(self):
+        result = {
+            '@context': 'http://schema.org',
+            '@type': 'Dataset',
+            '@id': str(self.id),
+            'description': mdstrip(self.description),
+            'alternateName': self.slug,
+            'dateCreated': self.created_at.isoformat(),
+            'dateModified': self.last_modified.isoformat(),
+            'url': url_for('datasets.show', dataset=self, _external=True),
+            'name': self.title,
+            'keywords': ','.join(self.tags),
+            'distribution': [resource.json_ld for resource in self.resources],
+            # This value is not standard
+            'extras': map(self.get_json_ld_extra, self.extras.items()),
+        }
+
+        if self.license and self.license.url:
+            result['license'] = self.license.url
+
+        if self.organization:
+            author = self.organization.json_ld
+        elif self.owner:
+            author = self.owner.json_ld
+        else:
+            author = None
+
+        if author:
+            result['author'] = author
+
+        return result
+
+    @staticmethod
+    def get_json_ld_extra((key, value)):
```

Comment:
I agree and, from a personal point of view, I prefer free function to class methods, since it is simpler and easier to test.

But:
- In the file, there are only classes, and this function is obviously related to a class. So, it made sense to 'attach' it to the class
-  I am ready to change it to a free function if we decide, as a coding guideline, that we should **always** use free functions instead of static methods.